api/opds.py

```diff
@@ -401,22 +407,37 @@ def add_author_links(self, work, feed, entry):
     def add_series_link(self, work, feed, entry):
         series_tag = OPDSFeed.schema_('Series')
         series_entry = entry.find(series_tag)
-
+        if not series_entry:
+            # There is no <series> tag, and thus nothing to annotate.
+            # This probably indicates an out-of-date OPDS entry.
+            if isinstance(work, Work):
+                work_id = work.id
+                work_title = work.title
+            else:
+                work_id = work.works_id
+                work_title = work.sort_title
+            self.log.error(
+                'add_series_link() called on work %s ("%s"), which has no <schema:Series> tag in its OPDS entry.',
+                work_id, work_title
+            )
+            return
```

Comment:
Yeah, since this should never happen I decided it was simpler to leave the entry alone rather than try to regenerate the entry on the spot.

If it does happen, there's no guarantee that regenerating the OPDS entry will fix the problem.